platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/bls/protocol/BlsProtocolManager.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.bls.protocol;
+
+import com.swirlds.common.system.NodeId;
+import com.swirlds.platform.bls.message.ProtocolMessage;
+import java.util.Collection;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+
+/**
+ * A manager for a BLS protocol
+ *
+ * @param <T> the type of output of the protocol being managed
+ */
+public interface BlsProtocolManager<T extends ProtocolOutput> {
+    /**
+     * Executes the next round of the protocol
+     *
+     * @param inputMessages the messages that are an input to the round
+     * @return the message generated from the protocol round
+     * @throws IllegalStateException if no more rounds exist to be called
+     */
+    ProtocolMessage executeNextRound(List<ProtocolMessage> inputMessages);
+
+    /**
+     * Finishes the protocol
+     *
+     * @param inputMessages the input messages needed to finish
+     * @return the protocol output object
+     */
+    T finish(List<ProtocolMessage> inputMessages);
+
+    /**
+     * Accepts a list of messages, filters them, and returns a list containing messages which match a certain subtype,
+     * cast to that subtype
+     * <p>
+     * Filters out messages that should be ignored by the protocol
+     *
+     * @param inputMessages        a list of messages of type U
+     * @param acceptableSenders    a list of senders from whom we are expecting to potentially receive messages
+     * @param subtype              the type of messages that should be returned
+     * @param disqualifyNonSenders if true, nodes included in the acceptableSenders parameter that didn't send a valid
+     *                             message will be declared offline
+     * @param <U>                  the type of messages that should be returned
+     * @return a list of messages of type U
+     */
+    <U extends ProtocolMessage> List<U> filterCast(
+            List<ProtocolMessage> inputMessages,
+            Collection<NodeId> acceptableSenders,
+            Class<U> subtype,
+            boolean disqualifyNonSenders);
```

Comment:
Is there any reason why this has to be in the interface? Could such a method be a static method somewhere?

Reply:
The reason is alluded to in the `disqualifyNonSenders` param doc - this function currently has the responsibility of disqualifying counterparties, based on the results of the filtering. The disqualification is closely intertwined with the internals of the `BlsProtocolManager`

I can update the main javadoc to make this more clear. If you're not a fan of how it's laid out, maybe it makes sense to reconsider once the implementation(s) of this are merging? (this iterative merge process can make parts of the review cumbersome, since the full picture is lacking)

Reply:
Ok, that makes sense. Let's leave this as-is for now.